### PR TITLE
Fixed #5676: Sum cost by quantity

### LINKED ISSUE #5676: Accessories/Components Total Value Ignores Quantity
#### Please confirm you have done the following before posting your bug report:

- [x] I have enabled debug mode 
- [x] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

**Describe the bug**
when accessing `/accessories` or `/components` the total value of all accessories, at the bottom of the table, is the sum of the Purchase Cost column, and does not take into account the Quantity column, I believe the correct total should be the sum of line totals, which would be caclulated by doing purchase cost * Qty. 

I believe this is a bug and not by design due to the way Purchase cost is used elsewhere in Snipe-IT, biggest example being Components checked out to Assets: on the `/hardware/1#components` page, Purchase Cost is shown (unmodified), regardless of how many are checked out to that asset, which makes sense, as you want to know the total value of the components assigned to that asset (related #2416 - please can this be reopened as the stale bot closed it without it being addressed).

**To Reproduce**
1. Go to '/accessories/create'
2. Create an Accessory with a quantity of 10 and a Purchase Cost of 10.00
3. Click on 'Save'
4. See the total at the bottom of the page shows 10.00 not 100.00.

**Expected behavior**
All Total fields which calculate their value based on the Purchase Cost of classes of items with a quantity should first multiple the Purchase Cost by Quantity before adding all values up.

**Screenshots**
accessoriestotal

**Server (please complete the following information):**
 - Snipe-IT Version **v4.4.1**
 - OS: [e.g. Ubuntu, CentOS] **Ubuntu**
 - Web Server: [e.g. Apache, IIS] **Apache**
 - PHP Version **v7.0.30**

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] **Windows 10**
 - Browser [e.g. chrome, safari] **Chrome**
 - Version [e.g. 22] **67**

**Smartphone (please complete the following information):**
 - Device: [e.g. iPhone6] **S8+**
 - OS: [e.g. iOS8.1] **Android 8.0.0**
 - Browser [e.g. stock browser, safari] **Chrome**
 - Version [e.g. 22] **Latest**

**Error Messages**
- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error **No error logged**
- If a stacktrace is provided in the error, include that too. **No error logged**
- Any errors that appear in your browser's error console. **No error logged**
- Confirm whether the error is reproducible on the demo: https://snipeitapp.com/demo. **Yes**
- Include any additional information you can find in `storage/logs` and your webserver's logs. **No error logged**

**Additional context**
- Is this a fresh install or an upgrade? **Upgrade**
- What OS and web server you're running Snipe-IT on **Ubuntu / Apache**
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc) **Recommended git clone method**
- Include what you've done so far in the installation, and if you got any error messages along the way. **install, configure and populate without errors**
- Indicate whether or not you've manually edited any data directly in the database **No**

Please do not post an issue without answering the related questions above. If you have opened a different issue and already answered these questions, answer them again, once for every ticket. It will be next to impossible for us to help you.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/59381819-accessories-components-total-value-ignores-quantity?utm_campaign=plugin&utm_content=tracker%2F505106&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F505106&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Computroniks**
# Description

Currently the total cost displayed on the components, accessories, consumables and licences does not take into account the quantity of the item. An new function has been created based upon the sumFormatter function that takes into account the quantity of an item when calculating the total cost. It takes the purchase cost and multiplies it by the quantity column before adding it to the total.

Fixes #5676 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

1. Create a new component, accessory, consumable or licence with a quantity of 1 and any purchase cost
2. Take note of the total at the bottom chrome_w9XviNtB8B
3. Increase the quantity of the item and note the total
chrome_BXEEyVtANl

**Test Configuration**:
* PHP version: 7.3.29-1~deb10u1
* MySQL version: 10.3.29-MariaDB-0+deb10u1
* Webserver version: Apache/2.4.38
* OS version: Raspbian GNU/Linux 10

# Checklist:

- [x] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [x] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Computroniks** [comment]
Fixed merge conflicts

**Comment by NikoGrano** [comment]
@snipe Sorry for pinging you directly here, but I've been following this and #5676 enough long without seeing any progress. 

Could it be possible to review this and merge if it's possible. Or give some comment about this at least.

**Comment by snipe** [comment]
@uberbrady has been doing a lot of work on the JS/backend for currency recently, so I'd need him to look at this before I'd be willing to merge.

**Comment by Computroniks** [comment]
Just looked at the commit for fixing the merge conflicts. Seems there is an issue with some of the lines getting duplicated. I will fix that now.

**Comment by Computroniks** [comment]
Fixed the merge conflicts.

**Comment by NikoGrano** [comment]
@uberbrady Would you mind to take look into this issue. Would be nice to get your input as there is active contributor to fix the problems - if any.

**Comment by Computroniks** [comment]
Anything I can do to help get this merged?

**Comment by uberbrady** [comment]
Yes - can you switch over to the European currency format? e.g. 1.234,56 - and see if it still works - and include screenshots?

We're slammed with the upcoming release of v6 and if you gave me something like that I might just take it and merge it, and deal with the consequences.

**Comment by Computroniks** [comment]
Ok, there was a slight issue when using the comma format. I've just switched out the parse used to convert purchase_cost to a float from parseFloat to cleanFloat and it now works. Here are the screenshots. Not all of the components can be seen in the screenshots so the numbers will appear to not add up. This is just because not everything can be seen. The important bit is that the two numbers match as we already know that the function works when we use the decimal format.

Decimal format:

Screenshot from 2021-10-30 22-49-38

and the comma format:

Screenshot from 2021-10-30 22-50-18

They match! :)

**Comment by Computroniks** [comment]
Fixed the merge conflicts

**Comment by NikoGrano** [comment]
Ping @uberbrady

File: resources/views/partials/bootstrap-table.blade.php [hunk 1]
```diff
@@ -653,6 +653,31 @@ function sumFormatter(data) {
...
+        if(Array.isArray(data)) {
+            
+            // Prevents issues on page load where data is an empty array
+            if(data[0] == undefined){
+                return 0.00
+            }
+            // Check that we are actually trying to sum cost from a table
+            // that has a quantity column. We must perform this check to
+            // support licences which use seats instead of qty
+            if('qty' in data[0]) {
```
Comment (uberbrady) [inline resources/views/partials/bootstrap-table.blade.php:666]: This is very clever, I love it!

**Review by uberbrady — APPROVED** [review]
Sorry for the delays - this looks really good to me, so we'll be happy to take it.

I particularly love the way of figuring out the `multiplier` - very nice :)

**Pull request merged**